lib/AppInfo/Application.php

```diff
@@ -224,56 +229,51 @@ public function boot(IBootContext $context): void {
 		ReferenceInvalidationListener::register($dispatcher);
 		ShareListener::register($dispatcher);
 
-		$context->injectFn(\Closure::fromCallable([$this, 'registerCloudFederationProviderManager']));
 	}
 
-	protected function registerCollaborationResourceProvider(IServerContainer $server): void {
-		/** @var IProviderManager $resourceManager */
-		$resourceManager = $server->get(IProviderManager::class);
+	public function registerCollaborationResourceProvider(IProviderManager $resourceManager, IEventDispatcher $dispatcher): void {
 		$resourceManager->registerResourceProvider(ConversationProvider::class);
-		$server->get(IEventDispatcher::class)->addListener(LoadAdditionalScriptsEvent::class, static function () {
+		$dispatcher->addListener(LoadAdditionalScriptsEvent::class, static function (): void {
 			Util::addScript(self::APP_ID, 'talk-collections');
 		});
 	}
 
-	protected function registerClientLinks(IServerContainer $server): void {
-		if ($server->getAppManager()->isEnabledForUser('firstrunwizard')) {
-			/** @var IManager $settingManager */
-			$settingManager = $server->getSettingsManager();
+	public function registerClientLinks(IAppManager $appManager, IManager $settingManager): void {
+		if ($appManager->isEnabledForUser('firstrunwizard')) {
 			$settingManager->registerSetting('personal', Personal::class);
 		}
 	}
 
-	protected function registerNavigationLink(IServerContainer $server): void {
-		$server->getNavigationManager()->add(static function () use ($server) {
-			/** @var Config $config */
-			$config = $server->get(Config::class);
-			$user = $server->getUserSession()->getUser();
+	public function registerNavigationLink(INavigationManager $navigationManager): void {
+		$navigationManager->add(static function () {
+			$config = Server::get(Config::class);
+			$userSession = Server::get(IUserSession::class);
+			$urlGenerator = Server::get(IURLGenerator::class);
+			$l = Server::get(IFactory::class)->get(self::APP_ID);
```

Comment:
Can't we inject these, too?

Reply:
Or maybe this is for performance, because we might not need them?

Reply:
exactly. will avoid setting up the stuff on all occ and api calls